tests/libs/estdlib/test_udp_socket.erl

```diff
@@ -0,0 +1,105 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2023 Fred Dushin <fred@dushin.net>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_udp_socket).
+
+-export([test/0]).
+
+test() ->
+    ok = test_echo_server(),
+    ok.
+
+test_echo_server() ->
+    Port = 44405,
+    {ok, ReceiveSocket} = socket:open(inet, dgram, udp),
+
+    ok = socket:setopt(ReceiveSocket, {socket, reuseaddr}, true),
+    ok = socket:setopt(ReceiveSocket, {socket, linger}, #{onoff => true, linger => 0}),
+
+    ok = socket:bind(ReceiveSocket, #{
+        family => inet, addr => loopback, port => Port
+    }),
+
+    Self = self(),
+    spawn(fun() ->
+        Self ! ready,
+        receive_loop(Self, ReceiveSocket)
+    end),
+
+    %% give receive_loop a chance to run
+    timer:sleep(100),
```

Comment:
This is different. I'm not convinced you need these waits, or ready message either.